Feature/async

aokigahara59 opened the pull request.

aokigahara59 committed:
Async-1 feat: rebuild client service for async working

aokigahara59 committed:
Async-2 feat: rebuild employee service for async work

aokigahara59 committed:
Async-3 feat: add async and Task complraison test

aokigahara59 committed:
Async-4 feat: reworked export and cash dispenser services

Dem-Ivan commented on Services.Tests/ThreadAndTaskTests.cs:
```diff
@@ -51,6 +52,60 @@ public void ParralelAddMoneyPositivTest()
         }
 
+        [Fact]
+        public async void AsyncTest()
```
Плохое имя.

Dem-Ivan commented on Services/ClientService.cs:
```diff
@@ -65,17 +66,18 @@ public List<Account> GetClientAccounts(Client client)
         }
 
-        public void UpdateAccount(Client client, Account oldAccount, Account newAccount)
+        public async Task UpdateAccount(Client client, Account oldAccount, Account newAccount)
```
Отметьте в названии метода что он асинхронный.

Dem-Ivan commented on Services/RateUpdater.cs:
```diff
@@ -21,7 +21,7 @@ public Task Start(CancellationToken cancelToken)
                         foreach (var account in accounts)
                         {
-                            clientService.UpdateAccount(client, account, new Account{Amount = account.Amount + 100});    
+                            await clientService.UpdateAccount(client, account, new Account { Amount = account.Amount + 100 });    
```
Ну и к какому выводу пришли? Лучше стало с await или хуже?

aokigahara59 committed:
Async-5 fix: Async services work fixes

Renamed services methods with prefix Async,
Added Update Account async/standart versions
Remade RateUpdater and CashDispenserServices

aokigahara59 committed:
Async-6 fix: fix return in cashdispencer service

aokigahara59 committed:
Async-7 fix: fixed parallel import export test
